src/WeakReference.mo

```diff
@@ -0,0 +1,30 @@
+/// Module that implements a weak reference to an object.
+///
+/// ATTENTION: This functionality does not work with classical persistence (`--legacy-persistence` moc flag).
+///
+/// Usage example:
+/// Import from the core package to use this module.
+/// ```motoko name=import
+/// import WeakReference "mo:core/WeakReference";
+/// ```
+
+import Prim "mo:⛔"
+
+module {
+  type WeakReference<T> = {
+    ref : weak T
+  };
+
+  public func allocateWeakRef<T>(obj : T) : WeakReference<T> {
+    return { ref = Prim.allocWeakRef<T>(obj) }
+  };
+
+  public func deref<T>(weakRef : WeakReference<T>) : ?T {
+    return Prim.weakGet<T>(weakRef.ref)
```

Comment:
Do we need the `WeakReference<T>` type, or would it be possible to simply pass `weak T` to the function (similar to `WeakReference.isLive`)? We probably want the input type to be the same for this function and `isLive`.

Reply:
you're right, they need to be the same -- but I ended up modifying the other one; since we promote using this module, then I guess we need to use it consistently, otherwise there's no need to hide the `weak T`.